### ISSUE TITLE
Output from expand_urls causes invalid Atom feeds

### ISSUE THREAD

**Issue by mikl** [imathis/octopress#198]
In `atom.xml`, the following line is used to escape and output the content:

```
    <content type="html">{{ post.content | expand_urls: site.url | xml_escape }}</content>
```

However, the output from expand_urls is not escaped, so if you have a code block in your code, you will get something like this:

``` html
&lt;p&gt;Mine looks like this:&lt;/p&gt;

&lt;p&gt;<div class='bogus-wrapper'><notextile><figure class='code'><figcaption><span> (solr.xml)</span> <a href='/downloads/code/solr.xml'>download</a></figcaption>
 <div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class='line-number'>1</span>
<span class='line-number'>2</span>
<span class='line-number'>3</span>
<span class='line-number'>4</span>
<span class='line-number'>5</span>
```

ie. with escaped and unescaped HTML mixed together.

As [the spec for text type decrees](http://www.atomenabled.org/developers/syndication/atom-format-spec.php#rfc.section.3.1.1.2), an element with `type="html"`must only contain escaped HTML. It is also allowed to set `type="xhtml" and then have unescaped markup instead, but then the whole thing must be unescaped and valid XHTML.

So, long story short, using code blocks or similar tags yields invalid Atom feeds.
